amoro-ams/src/main/java/org/apache/amoro/server/scheduler/inline/TableRuntimeRefreshExecutor.java

```diff
@@ -62,6 +63,16 @@ private void tryEvaluatingPendingInput(DefaultTableRuntime tableRuntime, MixedTa
     // only evaluate pending input when optimizing is enabled and in idle state
     if (tableRuntime.getTableConfiguration().getOptimizingConfig().isEnabled()
         && tableRuntime.getOptimizingStatus().equals(OptimizingStatus.IDLE)) {
+
+      if (tableRuntime.getTableConfiguration().getOptimizingConfig().isEventBasedTriggerEnabled()
+          && !MetricBasedEvaluationEvent.isEvaluatingNecessary(
+              tableRuntime.getOptimizingConfig(), table, tableRuntime.getLastPlanTime())) {
```

Comment:
what is the difference between 

`tableRuntime.getTableConfiguration().getOptimizingConfig()` and `tableRuntime.getOptimizingConfig()`, it looks confuse.